cylc/flow/rundb.py

```diff
@@ -1081,3 +1081,26 @@ def select_jobs_for_datastore(
     def vacuum(self):
         """Vacuum to the database."""
         return self.connect().execute("VACUUM")
+
+    def message_in_db(self, submit, message, timestamp, itask):
+        """Has this message been logged in the DB (task_events table)?
+
+        SQL Query in plain English:
+            Say whether the inner query returns any results:
+
+            If the event is a standard task message (e.g. Submit) then
+            there will be no message, so we chect the event column too.
+        """
+        task_name = itask.tokens['task']
+        cycle = itask.tokens['cycle']
+        stmt = r"""
+            SELECT EXISTS (
+                SELECT 1 FROM task_events WHERE
+                    name == ?
+                    AND cycle == ?
+                    AND (message == ? OR event == ?)
+                    AND submit_num == ?
+            )
+        """
+        stmt_args = [task_name, cycle, message, message, submit, timestamp]
+        return bool(self.connect().execute(stmt, stmt_args[:-1]).fetchone()[0])
```

Comment:
Timestamp no longer used? (Needs removing in the function definition + calls also)
```suggestion
        stmt_args = [task_name, cycle, message, message, submit]
        return bool(self.connect().execute(stmt, stmt_args).fetchone()[0])
```

Reply:
That shouldn't have happened - and I think that there's another problem I've not sorted out.